File: primes.hs
module Main (main) where
import GHC.Environment

primes = 2:3:[n | x<-[1..], y<-[-1,1], let n=6*x+y, isPrime n]
isPrime n = all (\x -> n `mod` x /= 0) $ takeWhile (\x -> x*x <= n) primes

getNumPrimesOrUseDefault :: [String] -> Int
getNumPrimesOrUseDefault [] = 100000
getNumPrimesOrUseDefault (x:xs) = 1000 * read x

sumPrimes :: Int -> Int
sumPrimes n = sum $ take n primes

main = do
    args <- getFullArgs
    let n = getNumPrimesOrUseDefault args
    print $ sumPrimes n

File: primes.py
import sys

def isPrime(n, plist):
    for p in plist:
        if n%p == 0: return False
        elif p*p > n: return True

def potentialPrimeGen():
    i=2
    while True:
        yield i//2 * 6 + i%2 * 2 - 1
        i+=1

def primes(nPrimes):
    primes = [0] * nPrimes
    primes[0] = 2
    primes[1] = 3
    idx = 2
    for p in potentialPrimeGen():
        if isPrime(p, primes):
            primes[idx] = p
            idx += 1
        if idx == nPrimes:
            return primes

def main():
    try:
        n = int(sys.argv[1]) * 1000
    except:
        n = 100000
    print sum(primes(n))

main()
